src/app/collective-rewards/shared/hooks/useMergeSeries.ts

```diff
@@ -1,39 +1,76 @@
-import { BackingPoint, RewardsPoint } from '@/app/collective-rewards/types'
+import { BackingPoint, CycleWindow, RewardsPoint } from '@/app/collective-rewards/types'
 import { useMemo } from 'react'
-import { toDate } from '@/app/collective-rewards/utils/chartUtils'
+import { ONE_DAY_IN_MS, ONE_DAY_IN_SECONDS } from '@/app/collective-rewards/constants/chartConstants'
+import { convertToTimestamp, formatDateKey } from '@/app/collective-rewards/utils/chartUtils'
+
+const calculateCycleDay = (date: Date, cycles: CycleWindow[]) => {
+  const targetTime = date.getTime()
+
+  for (const cycle of cycles) {
+    const startTime = convertToTimestamp(cycle.start)
+    const endTime = convertToTimestamp(cycle.end)
+
+    if (targetTime >= startTime && targetTime < endTime) {
+      const daysElapsed = Math.floor((targetTime - startTime) / ONE_DAY_IN_MS)
+      const displayDay = daysElapsed + 1
+
+      const cycleDurationDays = Math.ceil(cycle.cycleDuration / ONE_DAY_IN_SECONDS)
+
+      return {
+        cycle: cycle.cycleNumber || null,
+        dayInCycle: `${displayDay}/${cycleDurationDays}`,
+      }
+    }
+  }
+
+  return {
+    cycle: null,
+    dayInCycle: null,
+  }
+}
 
 type MergedDataPoint = {
   day: Date
   backing?: number
+  backingWei?: bigint
   rewardsUSD?: number
   rewardsRif?: number
   rewardsRbtc?: number
+  cycle?: number | null
+  dayInCycle?: string | null
 }
 
-export function useMergedSeries(backing: BackingPoint[], rewards: RewardsPoint[]) {
+export function useMergedSeries(backing: BackingPoint[], rewards: RewardsPoint[], cycles: CycleWindow[]) {
   return useMemo(() => {
```

Comment:
For this useMemo to work as expected I would add a useMemo from the place that their inputs comes from.